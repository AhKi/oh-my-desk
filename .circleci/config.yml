version: 2

jobs:
  build:
    working_directory: ~/oh-my-desk
    docker:
      - image: circleci/node:9.3.0
    environment:
      TZ: "/usr/share/zoneinfo/Asia/Seoul"
    steps:
      - checkout
      - restore_cache:
          key: dependency-cache-{{ .Branch }}-{{ checksum "yarn.lock" }}
      - run:
          name: Install Dependency
          command: yarn install
      - save_cache:
          key: dependency-cache-{{ .Branch }}-{{ checksum "yarn.lock" }}
          paths:
          - ./node_modules
          - ~/.yarn-cache
      - run:
          name: Test
          command: yarn test:coverage --maxWorkers=2
      - store_artifacts:
          path: coverage
          destination: coverage
      - run:
          name: Add Coverage Label
          command: yarn coveralls
