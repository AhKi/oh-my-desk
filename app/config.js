import { app, remote } from 'electron';
import path from 'path';

const BASIC_PATH = (app || remote.app).getPath('appData');
const TEMP_DIR = process.platform === 'win32' ? 'C:\\Windows\\Temp' : '/tmp';
const IS_TEST = process.env.NODE_ENV === 'test';
const SETTING_FILE = 'store.json';

export const SETTING_FILE_PATH = !IS_TEST ?
  path.join(BASIC_PATH, 'oh-my-desk', SETTING_FILE) :
  path.join(TEMP_DIR, SETTING_FILE);

const ROOT_PATH = path.resolve(__dirname, '..');

function getPagePath(target) {
  const ENV = process.env.NODE_ENV;
  const MIDDLE_PATH = ENV === 'development' ?
    path.join('app', 'renderer', 'pages', target) :
    path.join('build');

  return `file://${path.join(ROOT_PATH, MIDDLE_PATH, `${target}.html`)}`;
}
export const WIDGET_PATH = getPagePath('widget');
export const PREFERENCE_PATH = getPagePath('preference');
export const SEARCH_PATH = getPagePath('search');
export const UPDATE_WINDOW_PATH = getPagePath('UpdateWindow');
export const UPDATE_PROGRESS_PATH = getPagePath('UpdateProgress');

export const TRAY_ICON_PATH = path.join(ROOT_PATH, 'app', 'assets', 'iconTemplate.png');
export const LOGO_ICON_PATH = path.join(ROOT_PATH, 'app', 'assets', 'oh-my-desk-icon.png');

export const DEFAULT_SETTING = {
  identification: {
    widgetInfoById: {
      'fb07e56a-2ef3-4c3f-b7d4-3680a7173dd2': {
        size: {
          height: 787,
          width: 750,
        },
        isEditProgress: false,
        isMakeProgress: false,
        position: {
          x: 360,
          y: 83,
        },
        createTime: '2018-11-13T04:10:58.640Z',
        name: 'google',
        isOnTop: false,
        url: 'https://www.google.com/',
        isOpen: false,
        favorites: true,
        userAgent: 'DESKTOP',
        resentFocusTime: '2018-11-13T11:32:48.186Z',
        id: 'fb07e56a-2ef3-4c3f-b7d4-3680a7173dd2',
        reloadInterval: 5,
      },
      'e8581480-cd58-458a-a268-f9869fa5fcfc': {
        size: {
          height: 777,
          width: 821,
        },
        isEditProgress: false,
        isMakeProgress: false,
        position: {
          x: 575,
          y: 115,
        },
        createTime: '2018-11-13T11:32:08.049Z',
        name: 'trello',
        isOnTop: false,
        url: 'https://trello.com',
        isOpen: false,
        favorites: false,
        userAgent: 'DESKTOP',
        resentFocusTime: '2018-11-25T16:33:47.625Z',
        id: 'e8581480-cd58-458a-a268-f9869fa5fcfc',
      },
      'd5db9663-4879-47ca-ab63-10cf74de2967': {
        size: {
          height: 600,
          width: 500,
        },
        isEditProgress: false,
        isMakeProgress: false,
        position: {
          x: 600,
          y: 99,
        },
        createTime: '2018-11-25T16:34:16.673Z',
        name: 'translator',
        isOnTop: false,
        url: 'https://translate.google.com/',
        isOpen: false,
        favorites: false,
        userAgent: 'DESKTOP',
        resentFocusTime: '2018-11-25T16:34:17.266Z',
        id: 'd5db9663-4879-47ca-ab63-10cf74de2967',
      },
    },
  },
};
