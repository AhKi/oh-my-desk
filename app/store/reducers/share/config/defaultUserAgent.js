import { handleActions } from 'redux-actions';
import * as TYPES from 'actions/actionTypes';

const initialState = 'DESKTOP';

const DESKTOP = 'DESKTOP';
const MOBILE = 'MOBILE';

const defaultUserAgentReducer = handleActions({
  [TYPES.TOGGLE_WIDGET_MODE]: (state) => {
    if (state === MOBILE) {
      return DESKTOP;
    }

    return MOBILE;
  },
}, initialState);

export default defaultUserAgentReducer;
