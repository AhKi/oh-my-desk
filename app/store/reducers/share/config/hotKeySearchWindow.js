import { remote } from 'electron';
import { handleActions } from 'redux-actions';
import * as TYPES from 'actions/constant/actionTypes';

const initialState = 'Ctrl+Space';

const HotKeySearchWindowReducer = handleActions({
  [TYPES.SET_HOT_KEY_SEARCH_WINDOW]: (state, action) => {
    const { key } = action.payload;

    if (!remote) { // if main process of electron
      const handlingSearchHotKey = require('main/utils/menu/handlingSearchHotKey'); // eslint-disable-line
      handlingSearchHotKey(key);
    }

    return key;
  },
}, initialState);

export default HotKeySearchWindowReducer;
