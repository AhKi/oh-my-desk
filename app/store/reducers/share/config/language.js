import { handleActions } from 'redux-actions';
import * as TYPES from 'actions/constant/actionTypes';

const initialState = 'English';

const languageReducer = handleActions({
  [TYPES.SET_LANGUAGE_ENGLISH]: () => 'English',
  [TYPES.SET_LANGUAGE_KOREAN]: () => 'Korean',
}, initialState);

export default languageReducer;
