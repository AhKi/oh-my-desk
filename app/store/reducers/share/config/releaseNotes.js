import { handleActions } from 'redux-actions';
import * as TYPES from 'actions/constant/actionTypes';

const initialState = null;

const releaseNotesReducer = handleActions({
  [TYPES.UPDATE_CHECK_SUCCESS]: (state, action) => action.payload.releaseNotes,
}, initialState);

export default releaseNotesReducer;
