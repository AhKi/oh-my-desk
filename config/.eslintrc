{
  "extends": "airbnb",
  "rules": {
    "camelcase": 0,
    "react/prefer-stateless-function": 0,
    "jsx-a11y/href-no-hash": 0,
    "no-underscore-dangle": 0,
    "function-paren-newline": [
      "error",
      "consistent"
    ],
    "import/no-extraneous-dependencies": 0,
    "import/extensions": 0,
    "import/no-unresolved": 0,
    "import/no-named-as-default": 0,
    "import/no-duplicates": 0,
    "import/no-named-as-default-member": 0,
    "jsx-a11y/anchor-is-valid": 0,
    "linebreak-style": 0,
    "operator-linebreak": [2, "after"],
    "jsx-a11y/heading-has-content": 0,
    "react/jsx-one-expression-per-line": 0,
    "lines-between-class-members": 0,
    "no-continue": 0,
    "no-bitwise": 0,
    "react/destructuring-assignment": 0,
    "jsx-a11y/label-has-for": 0
  },
  "parser": "babel-eslint",
  "globals": {
    "__DEV__": false,
    "__TEST__": false,
    "__PROD__": false,
    "__API_ROOT__": false
  },
  "env": {
    "browser": true,
    "node": true,
    "jest": true
  }
}
