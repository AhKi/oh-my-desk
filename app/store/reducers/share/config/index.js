import { combineReducers } from 'redux-immutable';
import defaultUserAgent from './defaultUserAgent';
import hotKeySearchWindow from './hotKeySearchWindow';
import language from './language';

const configReducer = combineReducers({
  defaultUserAgent,
  hotKeySearchWindow,
  language,
});

export default configReducer;
