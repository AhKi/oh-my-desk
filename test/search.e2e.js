import testManager from './testManager';
import windowManager from './windowManager';
import * as config from './config';

jest.setTimeout(30000);

describe('test search window', async () => {
  const app = config.createApp();
  const browser = windowManager(app);
  const test = testManager(app);

  beforeAll(async () => {
    config.resetConfigFile();
    await app.start();
  });
  afterAll(async () => {
    await app.stop();
    config.resetConfigFile();
  });

  beforeEach(async () => {
    await app.client.waitUntilWindowLoaded();
  });

  it('start initial app', async () => {
    await test.matchWindowCount(2);
    await test.matchWindowTitle('oh-my-desk widget');
  });

  it('open search window and focus', async () => {
    await browser.searchWindowOpen();
    await test.matchWindowCount(3);
    await app.client.windowByIndex(2);
    await test.matchWindowTitle('oh-my-desk search');
    await test.matchValue('.SearchInput__input', '');
  });

  it('assert default widget list', async () => {
    const container = await test.getElement('.SearchList');
    const list = container.querySelectorAll('.SearchItem');
    const expectText = ['google', 'translator', 'trello'];

    list.forEach((item, index) => {
      expect(item.querySelector('strong').textContent)
        .toBe(expectText[index]);
    });
  });

  it('type search input', async () => {
    await test.matchValue('.SearchInput__input', '');
    await app.client.keys('trans');
    await test.matchValue('.SearchInput__input', 'trans');
  });

  it('assert searched widget list', async () => {
    const container = await test.getElement('.SearchList');
    const list = container.querySelectorAll('.SearchItem');
    expect(list.length).toBe(1);
    expect(list[0].querySelector('strong').textContent)
      .toBe('translator');
  });

  it('open widget using enter key', async () => {
    await app.client.keys('Enter');
    expect(await app.browserWindow.isVisible()).toBe(false);
    await app.client.pause(1000);
    await app.client.windowByIndex(4);
    await test.matchWindowCount(5);
    await test.matchWindowTitle('oh-my-desk widget');
    await test.matchText('.TitleBar__title', 'translator');
  });

  it('close widget using toggle btn', async () => {
    await browser.searchWindowOpen();
    await app.client.windowByIndex(2);
    await test.matchWindowTitle('oh-my-desk search');
    await test.matchValue('.SearchInput__input', '');
    expect(await app.browserWindow.isVisible()).toBe(true);

    let btn = await test.getElement('.ToggleButton__wrapper', 0);
    expect(btn.classList.contains('ToggleButton__wrapper-active'))
      .toBe(true);

    await app.client.click('.SearchItem__toggle-btn');
    await test.matchWindowCount(3);

    btn = await test.getElement('.ToggleButton__wrapper', 0);
    expect(btn.classList.contains('ToggleButton__wrapper-active'))
      .toBe(false);
  });

  it('open widget using toggle btn', async () => {
    await app.client.click('.SearchItem__toggle-btn');
    await app.client.pause(1000);
    await test.matchWindowCount(5);

    const btn = await test.getElement('.ToggleButton__wrapper', 0);
    expect(btn.classList.contains('ToggleButton__wrapper-active'))
      .toBe(true);
  });
});
