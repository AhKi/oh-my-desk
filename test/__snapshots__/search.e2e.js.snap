// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test search window feature of menu open new widget window 1`] = `
<div
  class="MakeNotice__bar"
>
  <p
    class="MakeNotice__bar-text"
  >
    This is register page of widget. Please complete register widget after searching URL that you want.
  </p>
  <button
    class="MakeNotice__bar-btn"
    type="button"
  >
    Make Widget &gt;
  </button>
</div>
`;
